File: aBeautifulYear.cpp
using namespace std;
#include <iostream>
#include <string>

bool isDistinct(int year) {
    string year_s = to_string(year);
    return (year_s[0] != year_s[1] && year_s[0] != year_s[2] && year_s[0] != year_s[3] && year_s[1] != year_s[2] && year_s[1] != year_s[3] && year_s[2] != year_s[3]);
}

int main() {
    int input;
    int year;
    cin >> input;
    year = input;
    do {
        year++;
    } while (!isDistinct(year));
    
    cout << year << endl;
    return 0;
}

File: aBeautifulYear.py
def isDistinct(year):
    year = str(year)
    return (year[0] != year[1] and year[0] != year[2] and year[0] != year[3] and year[1] != year[2] and year[1] != year[3] and year[2] != year[3])

year = int(input())
while True:
    year+=1
    if(isDistinct(year)):
       print (year)
       break
